sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/client.py

```diff
@@ -0,0 +1,194 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+try:
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = False
+
+if TYPE_CHECKING:
+    # pylint:disable=unused-import
+    from typing import Any, Optional, Union
+    from azure.core.credentials import TokenCredential
+    from . import EncryptionAlgorithm, KeyWrapAlgorithm, SignatureAlgorithm
+
+from azure.core.exceptions import HttpResponseError
+import six
+
+from . import DecryptResult, EncryptResult, SignResult, VerifyResult, UnwrapKeyResult, WrapKeyResult
+from ..models import Key
+from .._shared import KeyVaultClientBase, parse_vault_id
+
+
+class CryptographyClient(KeyVaultClientBase):
+    """
+    Performs cryptographic operations using Azure Key Vault keys.
+
+    :param key:
+        Either a :class:`~azure.keyvault.keys.models.Key` instance as returned by
+        :func:`~azure.keyvault.keys.KeyClient.get_key`, or a string.
+        If a string, the value must be the full identifier of an Azure Key Vault key with a version.
+    :type key: str or :class:`~azure.keyvault.keys.models.Key`
+    :param credential: An object which can provide an access token for the vault, such as a credential from
+        :mod:`azure.identity`
+
+    Keyword arguments
+        - *api_version* - version of the Key Vault API to use. Defaults to the most recent.
+    """
+
+    def __init__(self, key, credential, **kwargs):
+        # type: (Union[Key, str], TokenCredential, Any) -> None
+
+        if isinstance(key, Key):
+            self._key = key
+            self._key_id = parse_vault_id(key.id)
+        elif isinstance(key, six.text_type):
+            self._key = None
+            self._key_id = parse_vault_id(key)
+            self._get_key_forbidden = None  # type: Optional[bool]
+        else:
+            raise ValueError("'key' must be a Key instance or a key ID string including a version")
+
+        if not self._key_id.version:
+            raise ValueError("'key' must include a version")
+
+        super(CryptographyClient, self).__init__(vault_url=self._key_id.vault_url, credential=credential, **kwargs)
+
+    @property
+    def key_id(self):
+        # type: () -> str
+        """
+        The full identifier of the client's key.
+
+        :rtype: str
+        """
+        return "/".join(self._key_id)
+
+    @property
+    def key(self):
```

Comment:
Do you think that it's expected that this might make a network call given that it's a property?  would a method make more sense?

Reply:
No, yes, I'll make it `get_key`.